### ISSUE TITLE
Weird timeouts on test execution in fargate

### ISSUE THREAD

**Issue by bbeesley** [artilleryio/artillery#2811]
Version info: 
```
2.0.14
```

Running this command: 
```
artillery run-fargate --region eu-central-1 --count 5 --cluster api-core-eu-central-1-development --spot \
    --note "with cache" \
    --subnet-ids "`aws ec2 describe-subnets --filters 'Name=tag:Tier,Values=Public' --query 'Subnets[*].SubnetId' --output text | tr '\t' ','`" \
    --security-group-ids "`aws ec2 describe-security-groups --filters 'Name=group-name,Values=default' --query 'SecurityGroups[*].GroupId' --output text`" \
    --tags "environment:staging,actor:bbeesley,branch:main,runners:5" \
    -e staging comments_for_waveform.yml --record --key ***
```

I expected to see this happen:

A 14 minute long test should run from fargate and exit successfully, as it did with version 2.0.12, and has been doing successfully for a few weeks.

Instead, this happened:

Timeouts. Essentially the test never really stops. The throughput slowly drops down and down after the actual test has completed, but even an hour later metrics are still appearing:

```
--------------------------------------
Metrics for period to: 13:56:40(+0000) (width: 4.054s)
--------------------------------------

http.codes.200: ................................................................ 21
http.downloaded_bytes: ......................................................... 33841
http.request_rate: ............................................................. 8/sec
http.requests: ................................................................. 20
http.response_time:
  min: ......................................................................... 41
  max: ......................................................................... 3810
  mean: ........................................................................ 254.4
  median: ...................................................................... 76
  p95: ......................................................................... 144
  p99: ......................................................................... 144
```

At which point we start getting error messages

```
SQSError: SQS receive message failed: The security token included in the request is expired
```

Since the aws auth step only requests a role with a 60 min duration since we don't run tests as long as 60 mins.

Weirdly, if you look at the test result in artillery cloud it shows as [failed after 15 mins](https://app.artillery.io/load-tests/ty5pf_6x9kan7ze7zebgfnqjgxf9jkebjwj_bn8a).

Probably not quite related, but we used to use private subnets to run the tests from, but as of the latest version that has stopped working too. It's unclear from the docs that the requirement is to use public subnets (and it's unclear _why_ that would be a requirement), but if you use private subnets the fargate tasks fail to launch since they're configured to request a public ip address.

tldr; something fundamental seems to have changed in terms of fargate execution in the latest version and it's broken our ability to run tests in fargate.

**Comment by hassy** [comment]
*should* be fixed by #3190
